cartographer: bump to 0.3.0

cirocosta opened the pull request:
## What this PR does / why we need it

Cartographer just got [v0.3.0 released](https://github.com/vmware-tanzu/cartographer/releases/tag/v0.3.0), so, bump it here in TCE too.

in this PR, proposed changes:

- `vendir` the latest 0.3.0 release from `vmware-tanzu/package-for-cartographer`
- copy smoke tests from 0.2.2 to 0.3.0 (no change in the smoke tests)


## Which issue(s) this PR fixes

None


## Describe testing done for PR

1. brought up a kind cluster with kapp-controller installed
2. added to it the package and packagemetadata objects for both cert-manager and cartographer (from `./addons/packages/`)
3. ran the smoke test

```console
$ ./0.3.0/cartographer-test.sh
 Installing package 'cert-manager.community.tanzu.vmware.com'
...
 'PackageInstall' resource install status: ReconcileSucceeded

 Added installed package 'cert-manager'
...
 'PackageInstall' resource install status: ReconcileSucceeded

 Added installed package 'cartographer'
...

sleeping 1s to wait for configmap

NAME                  DATA   AGE
workload-test-basic   0      0s


...
Uninstalled package 'cartographer' from namespace 'default'
Uninstalled package 'cert-manager' from namespace 'default'
namespace "cartographer-12275" deleted

cartographer passed
```

Commit:
cartographer: bump to 0.3.0

- `vendir` the latest 0.3.0 release from `vmware-tanzu/package-for-cartographer`
- copy smoke tests from 0.2.2 to 0.3.0 (no change in the smoke tests)

Signed-off-by: Ciro S. Costa <ciroscosta@vmware.com>

Commit:
tests: run smoke tests for cartographer 0.3.0

with the inclusion of the new 0.3.0 package for cartographer, bump it up
to 0.3.0.

Signed-off-by: Ciro S. Costa <ciroscosta@vmware.com>

seemiller commented on addons/packages/cartographer/vendir.yml:
```diff
@@ -2,12 +2,12 @@ apiVersion: vendir.k14s.io/v1alpha1
 minimumRequiredVersion: 0.12.0
 directories:
-  - path: 0.2.2
+  - path: 0.3.0
     contents:
       - path: .
         githubRelease:
           slug: vmware-tanzu/package-for-cartographer
-          tag: v0.2.2
+          tag: v0.3.0
           disableAutoChecksumValidation: true
           assetNames:
           - "package.yaml"
```
@cirocosta Instead of replacing 0.2.2 with 0.3.0, can you add a new path for `0.3.0`? That way we can be sure to keep the 0.2.2 version in sync as well. You can refer to the [kpack vendir file](https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/kpack/vendir.yml) for an example of this.

cirocosta replied:
done - rebased on [3ecc430](https://github.com/vmware-tanzu/community-edition/pull/3923/commits/3ecc430d7d7b83dd2ce5eb86a01aa513eaa0f0bf)

seemiller commented on addons/packages/cartographer/0.3.0/package.yaml:
```diff
@@ -0,0 +1,29 @@
+apiVersion: data.packaging.carvel.dev/v1alpha1
+kind: Package
+metadata:
+  name: cartographer.community.tanzu.vmware.com.0.3.0
```
@cirocosta @kartiklunkad26 The name of the package and the imgpkgBundle.image field are different here than what is in the official 0.3.0 release of the package. Is this intentional?

I ran a `vendir sync` in the `addons/packages/cartographer` directory and got differences in the 0.3.0 package.yaml file.

The name in the PR is `cartographer.community.tanzu.vmware.com.0.3.0` while in the release it is `cartographer.tanzu.vmware.com.0.3.0`. The image also changes location from TCE's Harbor to Docker.

Can you confirm what should be here?

cirocosta replied:
@seemiller good catch! 

I just updated the PR with an updated asset structure for the releases we generate in `vmware-tanzu/package-for-cartographer` to better deal with handling multiple `cartographer-*` projects from there (described the problem here: https://github.com/vmware-tanzu/package-for-cartographer/issues/27)

with those changes (mentioned in the issue above) there'll be no further mismatches between `vendir sync`s here.

---

just to be sure: would it be possible to make use of `cartographer.tanzu.vmware.com` and images coming from dockerhub (actually, we'll be moving our public images to `ghcr.io` soon)? that'd certainly simplify some of the work there in `package-for-cartographer` 😅  specially the automation as (pls correct me if I'm wrong) we're not able to push to the registry from outside VMware infra.  (if we should stick with producing `cartographer.community.tanzu.vmware.com` and pushing to `projects.registry.tanzu...` that's fine, it'd just be easier the other way 😁 )

thank you!

seemiller replied:
@cirocosta Names and OCI registries are up to your team's and organization's guidelines. I am ok with the name that you chose. And I believe that VMware would prefer our Harbor registry. However, our package policy does not specifically exclude others like GCR, GHCR or Docker Hub..

Commit:
cartographer: consume release contents from tgz

given that the repository where does bits come from
(vmware-tanzu/package-for-cartographer) is aimed towards the generation
of packages for multiple projects targetted at multiple distributions
(e.g., TAP _and_ TCE), now for each projects and distribution, a `.tgz`
is included in the release so that `package.yaml` for each can be
fetched.

note that previous releases (0.2.2) have been updated to conform to the
same pattern.

Signed-off-by: Ciro S. Costa <ciroscosta@vmware.com>

seemiller commented:
Package repo for testing:
```
projects-stg.registry.vmware.com/tce/main:v0.12.0-alpha.3923.1
```
